### PR TITLE
[DOC] Setup

### PR DESCRIPTION
**Pull request by eseiler**
Maybe we should do the directory basically the same way we do the snippets/solutions of the manual?
That is:

- Have one subdirectory per section
- Have CMake fetch all cpp files and prefix them with the section (unique target name)

### CODE REVIEW

File: doc/setup/index.md [hunk 1]
```diff
@@ -1 +1,169 @@
...
+
+```bash
+cmake ../source
+make
+./hello_world
+```
+
+The output should be `Hello world`.
+
+\note Depending on the standard C++ on your system, you may need to specify the compiler via `-DCMAKE_CXX_COMPILER=`, e.g. `-DCMAKE_CXX_COMPILER=g++-7` or the full path to the executable.
```
Comment (smehringer) [inline doc/setup/index.md]:
```suggestion
\note Depending on the standard C++ on your system, you may need to specify the compiler via `-DCMAKE_CXX_COMPILER=`, for example:
```bash
cmake ../source -DCMAKE_CXX_COMPILER=/path/to/executable/g++-7`
\```
```

File: doc/setup/index.md [hunk 2]
```diff
@@ -1 +1,169 @@
...
+
+<!--
+<h4 id='setup--what-happens'>What happens</h4>
```
Comment (smehringer) [inline doc/setup/index.md]:
```suggestion
What happens {#section_what_happens}
```
can be references than: `[Here](#section_what_happens)`

File: doc/setup/index.md [hunk 3]
```diff
@@ -1 +1,169 @@
...
+
+We use cmake as out build software.
```
Comment (smehringer) [inline doc/setup/index.md]:
```suggestion
We use cmake as our build software.
```

File: doc/setup/index.md [hunk 4]
```diff
@@ -1 +1,149 @@
...
+
+**Ubuntu >= 18.04**
+```
+sudo apt-get install gcc
+```
+**Ubuntu < 18.04**
+```
+sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+sudo apt-get update
+sudo apt-get install gcc-7
```
[the remark this answers is not in the document]
Comment (h-2) [reply doc/setup/index.md]: also `apt` is recommended over `apt-get` nowadays, I think.

File: doc/setup/index.md [hunk 5]
```diff
@@ -1 +1,149 @@
...
+
+\note If you encounter the error <code>/usr/lib/x86_64-linux-gnu/libstdc++.so.6: version 'CXXABI_1.3.11' not found</code>, you have to set the LD_LIBRARY_PATH:
+```
+export LD_LIBRARY_PATH=/home/user/miniconda3/envs/gcc7/lib/
+```
+where `/home/user/miniconda3/` is the path to your conda installation.
+
+\attention After installing, `g++ --version` should print the desired version. If not, you may have to use, for example, `g++-7 --version` or even specify the full path to your executable.
+
+Similarly you need to install cmake and git.
```
Comment (h-2) [inline doc/setup/index.md]: Maybe make just one command line that they can copy'n'paste, like:
```bash
apt install g++-7 cmake git
```

**Pull request merged**